com.seventhstring.Transcribe.yml

```diff
@@ -0,0 +1,62 @@
+app-id: com.seventhstring.Transcribe
+runtime: org.freedesktop.Platform
+runtime-version: '21.08'
+sdk: org.freedesktop.Sdk
+command: transcribe
+rename-desktop-file: seventhstring-transcribe.desktop
+rename-icon: seventhstring-transcribe
+finish-args:
+#TODO: figure out if its possible to only grant permissions 
+# on ~/.Transcribe!7 and xdg-music:ro, for now we use home
+# as a functional compromise
```

Comment:
It would be great to figure this out before we publish it on Flathub. You can change it to '--persist=.Transcribe!7` and Flatpak will create proper bind-mount.

Reply:
Hey, i'll try it today after work!

Reply:
That seems to mount a directory called `.Transcribe!7` into the container. After entering the license key it looks like this on my machine: 
```
mhoffm@fedora:~$ flatpak enter com.seventhstring.Transcribe ls -lha
total 8.0K
drwx------. 6 mhoffm mhoffm  160 Dec 22 16:22  .
drwx------. 3 mhoffm mhoffm   60 Dec 22 16:22  ..
drwxr-xr-x. 1 mhoffm mhoffm    0 Dec 19 12:17 '.Transcribe!7'
-rwxr-xr-x. 1 mhoffm mhoffm 2.0K Dec 22 16:22 '.Transcribe!7j3O7ay'
drwxr-xr-x. 3 mhoffm mhoffm   60 Dec 22 16:22  .local
drwxr-xr-x. 3 mhoffm mhoffm   60 Dec 22 16:22  .var
-rw-r--r--. 1 mhoffm mhoffm   39 Dec 22 16:22  .yeqwqauh
drwxr-xr-x. 1 mhoffm mhoffm  120 Dec 13 18:28  Music
```

The application seems to create a file with some random suffix if the filename is already present. Is it possible to use files for `--persist`?

Reply:
Is it possible? I found this: https://github.com/flatpak/flatpak/blob/412c15772f794fd300991a91f2e0fe51f615bf79/common/flatpak-context.c#L2555 

It seems to me that the argument of `--persist` will always appear as a directory inside the container.

Reply:
Yeah, that's the case. Is `.Transcribe!7` supposed to be a text file? If yes, I would suggest looking for it in some kind of subdirectory (e.g. `.config/Transcribe/Transcribe!7`) instead. This way, if application respects `XDG_CONFIG_HOME` environment variable, there will be no conflicts.

There's also `--filesystem:path:create`, but it will also create a directory and not a file.

Reply:
Its a file yes! Its also not configurable, and the application does not respect XDG_CONFIG_HOME. After trying and playing around with it, I dont think there is a better way really.

Reply:
It would be possible to provide a wrapper script to work around these limitations, but as it probably doesn't use XDG portals anyway, let's keep it with `--filesystem=home`.

Reply:
ok i utilized a wrapper script and that works! thanks